ext/phar/tests/zip/phar_buildfromiterator7.phpt

```diff
@@ -28,7 +28,7 @@ class myIterator implements Iterator
     }
     function valid(): bool {
         echo "valid\n";
-        return current($this->a);
+        return current($this->a) == true;
```

Comment:
```suggestion
        return current($this->a) !== false;
```

Reply:
I mainly converted it to `== true` in all the phar iterator tests because it is closest to the previous behavior and how implicit conversions to bool work, because I wanted to keep the behavior as much as possible like before. This pattern is also not just in this file, it is repeated 3x for buildfromiterator4, 6 and 7 and for tar, zip and the base test directory, so this occurs 9x. I am not opposed to changing this, but I am also not sure it improves anything.

Reply:
I understand your intention, but when changing it, IMHO it is better to change it to a form a developer would choose /w this PR alredy implemented. ` == true` is ultra hard to read.

Reply:
I get your point. I will replace the `== true` with `!== false` for all the iterators and check the other test fixes if there is a way to make the adaptions more concise. Thanks!